src/main/kotlin/de/hysky/skyblocker/skyblock/SlotSwap.kt

```diff
@@ -0,0 +1,182 @@
+	private var shouldResetOnNextKey = false
+	private var waitJob: Job? = null
+	private const val WAIT_TIME = 500L
+
+	private const val SLOT_SIZE = 16
+	private const val HALF_SLOT_SIZE = SLOT_SIZE / 2
+
+	//These 2 aren't configurable, just because
+	private const val CONFIGURING_SOURCE_SLOT_COLOR = 0xFF993AFF.toInt()
+	private const val CONFIGURING_TARGET_SLOT_COLOR = 0xFFFF9214.toInt()
+	private const val LINE_WIDTH = 2f
+	private const val OFFSET_TO_CENTER = 1f // This is needed because SlotSquares' x and y values are 1 less than what they should be for being visually in the center.
+
+	@JvmStatic
+	fun init() {
+		ScreenEvents.AFTER_INIT.register { client, screen, _, _ ->
+			//Since this event is called for each screen, we can just check the config enabled status here once and for all
+			if (screen !is InventoryScreen || !config.enableSlotSwap) return@register
+			currentScreen = screen
+			ScreenKeyboardEvents.afterKeyPress(screen).register { _, key, scancode, _ ->
+				onKeyPress(screen, key, scancode)
+			}
+			ScreenKeyboardEvents.afterKeyRelease(screen).register { _, key, scancode, _ ->
+				onKeyRelease(client, screen, key, scancode)
+			}
+			ScreenEvents.afterRender(screen).register { _, drawContext, _, _, _ ->
+				render(screen, drawContext)
+			}
+			ScreenMouseEvents.allowMouseClick(screen).register { _, _, _, button ->
+				onMouseClick(client, screen, button)
+			}
+			ScreenEvents.remove(screen).register {
+				currentScreen = null
+			}
+		}
+	}
+
+	private fun onMouseClick(client: MinecraftClient, screen: InventoryScreen, button: Int): Boolean {
+		if (button != GLFW.GLFW_MOUSE_BUTTON_LEFT || !InputUtil.isKeyPressed(client.window.handle, GLFW.GLFW_KEY_LEFT_SHIFT)) return true
+
+		val entry: IntIntImmutablePair = (screen as HandledScreenAccessor).focusedSlot?.id?.let { id ->
+			slotMap.firstOrNull { pair -> pair.keyInt() == id || pair.valueInt() == id }
+		} ?: return true
+
+		if (isHotbarSlot(entry.keyInt())) {
+			client.interactionManager?.clickSlot(screen.screenHandler.syncId, entry.valueInt(), entry.keyInt() - 36, SlotActionType.SWAP, client.player)
+			return false
+		} else if (isHotbarSlot(entry.valueInt())) {
+			client.interactionManager?.clickSlot(screen.screenHandler.syncId, entry.keyInt(), entry.valueInt() - 36, SlotActionType.SWAP, client.player)
+			return false
+		}
+		return true
+	}
+
+	private fun onKeyPress(screen: InventoryScreen, key: Int, scancode: Int) {
+		if (!configureKeybinding.matchesKey(key, scancode) || slotAtKeyPress != null) return
+		slotAtKeyPress = screen.getFocusedSlotId()
+	}
+
+	private fun onKeyRelease(client: MinecraftClient, screen: InventoryScreen, key: Int, scancode: Int) {
+		when {
+			configureKeybinding.matchesKey(key, scancode) -> {
+				val slotAtKeyRelease = screen.getFocusedSlotId()
+				if (slotAtKeyPress != null && slotAtKeyRelease != null && slotAtKeyPress != slotAtKeyRelease
+					&& (isHotbarSlot(slotAtKeyPress!!) || isHotbarSlot(slotAtKeyRelease))
```

Comment:
It can be `null` if the key press happened while the cursor wasn't on a proper slot as you can see from `InventoryScreen.getFocusedSlotId()`. If `focusedSlot` is `null,` the whole expression evaluates to `null`. If it's not `null`, and then the slot is not valid for slot swap (not a part of inventory of armor slots), it's still going to end up as `null`.